src/Npgsql.GeoJSON/NpgsqlGeoJSONExtensions.cs

```diff
@@ -34,4 +34,31 @@ public static INpgsqlTypeMapper UseGeoJson(this INpgsqlTypeMapper mapper, CrsMap
         mapper.AddTypeInfoResolverFactory(new GeoJSONTypeInfoResolverFactory(options, geographyAsDefault, crsMap));
         return mapper;
     }
+
+    /// <summary>
+    /// Sets up GeoJSON mappings for the PostGIS types.
+    /// </summary>
+    /// <param name="mapper">The type mapper to set up (global or connection-specific)</param>
+    /// <param name="options">Options to use when constructing objects.</param>
+    /// <param name="geographyAsDefault">Specifies that the geography type is used for mapping by default.</param>
+    public static TMapper UseGeoJson<TMapper>(this TMapper mapper, GeoJSONOptions options = GeoJSONOptions.None, bool geographyAsDefault = false)
```

Comment:
Should we actually leave the non-generic just above, is that for binary compat? It actually looks pretty ambiguous between the two in terms of overload resultion (I'm assuming the generic is resolved, otherwise this is useless?). Am assuming it's not even worth obsoleting the above since it's not resolved by the compiler?

Reply:
Yes it gets picked, it's indeed mostly for binary compat, we could decide it's worth removing the original overloads.

The reason they're not obsolete is that for NpgsqlConnection.GlobalTypeMapper we still pick the exact match, which would cause perfectly fine code to start warning.

So we either remove the old methods or keep it as is.

Reply:
OK to keep as-is, maybe add a note on the old methods that they're for binary-compat only, so that we understand why they're still there in the future...